lib/object_3d.dart

```diff
@@ -296,10 +335,18 @@ class _ObjectPainter extends CustomPainter {
     List<Offset> offsets = [];
     List<Color> colors = [];
     for (int i = 0; i < faces.length; i++) {
-      List<int> face = faces[avgOfZ[i].index];
-      final n = _normalVector(verticesToDraw, face);
-      colors.addAll(_calcColor(color, n));
-      offsets.addAll(_drawFace(verticesToDraw, face));
+      final List<int> faceIdx = faces[avgOfZ[i].index];
+
+      // Fallback on default color func if a custom one is not provided
+      Face face = Face(verticesToDraw[faceIdx[0] - 1],
+          verticesToDraw[faceIdx[1] - 1], verticesToDraw[faceIdx[2] - 1]);
```

Comment:
I think it would be a bit nicer to extract these function parameters above this call so its a bit more readable

```suggestion
      final v1 = verticesToDraw[faceIdx[0] - 1];
      final v2 = verticesToDraw[faceIdx[1] - 1];
      final v3 = verticesToDraw[faceIdx[2] - 1];
      Face face = Face(v1, v2, v3);
```

Reply:
done